frontend/ui/presets.lua

```diff
@@ -0,0 +1,338 @@
+            callback = function()
+                preset_object.loadPreset(presets[preset_name])
+                print("XXX", presets[preset_name], preset_object, preset_name)
+                -- There's no guarantee that it'll be obvious to the user that the preset was loaded so, we show a notification.
+                UIManager:show(InfoMessage:new{
+                    text = T(_("Preset '%1' loaded successfully."), preset_name),
+                    timeout = 2,
+                })
+            end,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(ConfirmBox:new{
+                    text = T(_("What would you like to do with preset '%1'?"), preset_name),
+                    icon = "notice-question",
+                    ok_text = _("Update"),
+                    ok_callback = function()
+                        UIManager:show(ConfirmBox:new{
+                            text = T(_("Are you sure you want to overwrite preset '%1' with current settings?"), preset_name),
+                            ok_callback = function()
+                                presets[preset_name] = preset_object.buildPreset()
+                                preset_object:save()
+                                UIManager:show(InfoMessage:new{
+                                    text = T(_("Preset '%1' was updated with current settings"), preset_name),
+                                    timeout = 2,
+                                })
+                            end,
+                        })
+                    end,
+                    other_buttons_first = true,
+                    other_buttons = {
+                        {
+                            {
+                                text = _("Delete"),
+                                callback = function()
+                                    UIManager:show(ConfirmBox:new{
+                                        text = T(_("Are you sure you want to delete preset '%1'?"), preset_name),
+                                        ok_text = _("Delete"),
+                                        ok_callback = function()
+                                            presets[preset_name] = nil
+                                            preset_object:save()
+                                            local action_key = preset_object.dispatcher_name
+                                            if action_key then
+                                                UIManager:broadcastEvent(Event:new("DispatcherActionValueChanged", {
+                                                    name = action_key,
+                                                    old_value = preset_name,
+                                                    new_value = nil -- delete the action
+                                                }))
+                                            end
+                                            on_updated_callback()
```

Comment:
It's not good to rebuild the whole menu when you can just `table.remove` one item.

You laugh that I optimize "count to 2". We write thousands of lines of code and every optimized bit is important.
So you can continue laughing and I'll continue optimizing.